### PR TITLE
improve chain derivation documentation

### PR DESCRIPTION
**Pull request by roberto-bayardo**
**Description**

Updates the chain derivation docs to improve clarity & precision, fix errors, and correct typos.

## Summary by CodeRabbit

- **Documentation**
  - Enhanced the clarity and structure of the L2 chain derivation documentation.
  - Introduced a new section on "Eager Block Derivation" with detailed explanations.
  - Clarified terminology and process for mapping L1 blocks to L2 blocks.
  - Updated constraints on L2 block timestamps for better understanding.

### CODE REVIEW

File: specs/derivation.md [hunk 1]
```diff
@@ -93,77 +94,83 @@
...
+multiple L2 blocks. Each L2 block belongs to exactly one epoch, and we call the corresponding L1
+block its [L1 origin][l1-origin]. The epoch's number equals that of its L1 origin block.
+
+To derive the L2 blocks of epoch number `E`, we need the following inputs:
+
+- L1 blocks in the range `[E, E + SWS)`, called the [sequencing window][g-sequencing-window] of the epoch, and `SWS`
+  the sequencing window size. (Note that sequencing windows overlap.)
+- [Batcher transactions][g-batcher-transaction] from blocks in the sequencing window.
+  - These transactions allow us to reconstruct the epoch's [sequencer batches][g-sequencer-batch], each of
+    which will produce one L2 block. Note that:
```
Comment (0x00101010) [inline specs/derivation.md:112]: I think after SpanBatch, a batch can be used to build one or more blocks

File: specs/derivation.md [hunk 2]
```diff
@@ -93,77 +94,83 @@
...
-  - The L1 block attributes from L1 block `E` (to derive the [L1 attributes deposited transaction][g-l1-attr-deposit]).
-- The state of the L2 chain after the last L2 block of epoch `E - 1`, or — if epoch `E - 1` does not exist — the
-  [L2 genesis state][g-l2-genesis].
-  - An epoch `E` does not exist if `E <= L2CI`, where `L2CI` is the [L2 chain inception][g-l2-chain-inception].
-
-To derive the whole L2 chain from scratch, we simply start with the [L2 genesis state][g-l2-genesis], and the [L2 chain
-inception][g-l2-chain-inception] as first epoch, then process all sequencing windows in order. Refer to the
+The L2 chain is derived from the L1 chain. In particular, each L1 block following [L2 chain
+inception][g-l2-chain-inception] is mapped to a [sequencing epoch][g-sequencing-epoch] comprising
+multiple L2 blocks. Each L2 block belongs to exactly one epoch, and we call the corresponding L1
```
Comment (trianglesphere) [inline specs/derivation.md]:
```suggestion
at least one L2 block. Each L2 block belongs to exactly one epoch, and we call the corresponding L1
```
Typically multiple, but not strictly

File: specs/derivation.md [hunk 3]
```diff
@@ -93,77 +94,83 @@
...
-See [Batch Queue] for more details.
+The L2 chain may contain pre-Bedrock history, but the L2 genesis here refers to the Bedrock L2
+genesis block.
 
-## Eager Block Derivation
+Each L2 `block` with origin `l1_origin` is subject to the following constraints (whose values
+are denominated in seconds):
+
+- `block.timestamp = prev_l2_timestamp + l2_block_time`
+  - `prev_l2_timestamp` is the timestamp of the L2 block immediately preceeding this one, or
```
Comment (trianglesphere) [inline specs/derivation.md]: the block's timestamp is always `prev_l2_timestamp + l2_block_time`. The rules around what a valid L1 origin relate the various L1 and L2 timestamps together.

File: specs/derivation.md (hunk 1 above)
Comment (trianglesphere) [reply specs/derivation.md:112]: That is correct. We will need to draw the distinction between a singular batch and a span batch here. A second thing that I'm thinking of is that we should figure out how span batches relate the the more complex but rarely hit batch rules.

**Review by trianglesphere — COMMENTED** [review]
Thanks for the cleanup here. There's a couple small things that I think need to be fixed, but overall it reads much more clearly than it did.

File: specs/derivation.md (hunk 1 above)
Comment (roberto-bayardo) [reply specs/derivation.md:112]: Do we want to update this now or wait until span batches are live? (I suspect there will be more edits than this to cover span batches).

File: specs/derivation.md (hunk 3 above)
Comment (roberto-bayardo) [reply specs/derivation.md]: ack. any edits you'd like here to further clarify?

File: specs/derivation.md (hunk 1 above)
Comment (trianglesphere) [reply specs/derivation.md:112]: Span batches are spec'd, so we should start on this, but it should be part of a separate PR.

File: specs/derivation.md (hunk 3 above)
Comment (trianglesphere) [reply specs/derivation.md]: The part "or  `l1_origin.timestamp` if this is the first L2 block in the first epoch." is incorrect.

Comment (roberto-bayardo) [reply specs/derivation.md]: Ah of course: updated the text to clarify that the genesis block timestamp is explicitly provided.

**Review by trianglesphere — APPROVED** [review]
ty for cleaning this up @roberto-bayardo

**Pull request merged**